proposals/left-right-join-in-query-expressions.md

```diff
@@ -0,0 +1,235 @@
+# `left join` and `right join` query expression clauses
+
+Champion issue: https://github.com/dotnet/csharplang/issues/8947
+
+## Summary
+[summary]: #summary
+
+Introduce `left join` and `right join` clauses into LINQ query expression syntax, e.g.:
+
+```c#
+from student in students
+left join department in departments on student.DepartmentID equals department.ID
+select new { student.Name, department?.Name }
+```
+
+## Motivation
+[motivation]: #motivation
+
+LINQ has a [Join operator](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.join), which, like its SQL INNER JOIN counterpart, correlates elements of two sequences based on matching keys; the C# language has a corresponding `join` clause which translates to this operator ([section 12.20.3.5 of the C# specs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#122035-from-let-where-join-and-orderby-clauses)).
+
+In addition to INNER JOIN, SQL also has LEFT JOIN, which returns outer elements even if there's no corresponding inner ones; LINQ and C#, in contrast, lack this operator. [The LINQ conceptual documentation shows how to combine existing operators to achieve a left join](https://learn.microsoft.com/en-us/dotnet/csharp/linq/standard-query-operators/join-operations#perform-left-outer-joins):
```

Comment:
consider adding a section discussing and linking to the runtime work to add the new LeftJoin and RightJoin methods to linq, and how they are otherwise identical in shape to the existing join apis (link to those too).

Reply:
Will address this below.